inc/functions_upload.php

```diff
@@ -840,79 +840,104 @@ function add_attachments($pid, $forumpermissions, $attachwhere, $action=false)
 }
 
 /**
- * Delete an uploaded file both from the relative path and the CDN path if a CDN is in use.
+ * Delete an uploaded file both from the provided absolute path
+ * as well as from the associated CDN path if a CDN is in use.
  *
- * @param string $path The relative path to the uploaded file.
+ * @param string $abs_path The absolute path to the uploaded file.
+ *                         Must resolve to a directory under MYBB_ROOT.
  *
  * @return bool Whether the file was deleted successfully.
  */
-function delete_uploaded_file($path = '')
+function delete_uploaded_file($abs_path)
 {
 	global $mybb, $plugins;
 
 	$deleted = false;
 
-	$deleted = @unlink($path);
-
-	$cdn_base_path = rtrim($mybb->settings['cdnpath'], '/');
-	$path = ltrim($path, '/');
-	$cdn_path = realpath($cdn_base_path . '/' . $path);
-
-	if(!empty($mybb->settings['usecdn']) && !empty($cdn_base_path))
+	$realpath = realpath($abs_path);
+	if (substr($realpath, 0, strlen(MYBB_ROOT)) !== MYBB_ROOT || $realpath === realpath(MYBB_ROOT.'install/lock'))
 	{
-		$deleted = $deleted && @unlink($cdn_path);
+		return false;
 	}
+	else
+	{
+		$deleted = @unlink($realpath);
 
-	$hook_params = array(
-		'path' => &$path,
-		'deleted' => &$deleted,
-	);
+		$rel_path = substr($realpath, strlen(MYBB_ROOT));
+
+		if (!empty($mybb->settings['usecdn']))
+		{
+			$cdn_base_path = rtrim($mybb->settings['cdnpath'], '/');
+			if (!empty($cdn_base_path))
+			{
+				$cdn_path = realpath("{$cdn_base_path}/{$rel_path}");
+				$deleted = $deleted && @unlink($cdn_path);
+			}
+		}
+
+		$hook_params = array(
+			'path' => &$rel_path,
```

Comment:
What is the implication for plugins here ? @lairdshaw 

Would it be such that delaying this change for 1.9 is to be considered ?

Reply:
As far as I can see, the only implication is that if plugins have already been adapted to the original change from relative paths to absolute paths being supplied to this function - which (I thought was what) broke deletion of CDN files - then they will have to revert their adaptation. Otherwise, the supplied arguments will be identical.

(Caveat: I haven't looked into this code since first sharing it, and, as I said at the time, I didn't test it much even then.)